beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -300,8 +300,8 @@ where
         }
     }
 
-    /// Produce an `AttestationData` that is valid for the present `slot` and given `shard`.
-    pub fn produce_attestation_data(&self, shard: u64) -> Result<AttestationData, Error> {
+    /// Propose an `AttestationData` that is valid for the present `slot` and given `shard`.
+    pub fn propose_attestation_data(&self, shard: u64) -> Result<AttestationData, Error> {
```

Comment:
I think we should keep all the methods on the `BeaconChain` as "produce".

The work-flow is:

1. Validator asks a beacon node to _produce_ a block.
2. Beacon node _produces_ a block.
3. Validator approves that block, chooses to sign it and _propose_ it as a new block.
4. Validator sends new block _proposal_ to the beacon node to _publish_ it on the network.